src/main/java/org/cbioportal/web/IndexPageController.java

```diff
@@ -51,8 +56,28 @@ private Map<String, String> getFrontendProperties(HttpServletRequest request, Au
         properties.put("user_email_address", authentication != null ? authentication.getName(): "anonymousUser");
         // TODO: Support skin.user_display_name 
         properties.put("user_display_name", authentication != null ? authentication.getName(): "anonymousUser");
+        // Set MSK slide viewer token at runtime
+        properties.put("mskWholeSlideViewerToken", getMskWholeSlideViewerToken(env.getProperty("msk.whole.slide.viewer.secret.key"), authentication));
```

Comment:
I would suggest changing this to 
`
    @Value("${msk.whole.slide.viewer.secret.key:}")
    private String wholeSlideViewerKey;`